aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java

```diff
@@ -826,127 +829,136 @@ private void openURL(final String url, final boolean noSchemeValidation) {
+        String intentAction = notificationDetails.getIntentAction();
 
-        final String imageUrl = data.getString(CAMPAIGN_IMAGE_PUSH_KEY);
-        final String imageIconUrl = data.getString(CAMPAIGN_IMAGE_ICON_PUSH_KEY);
-        final String imageSmallIconUrl = data.getString(CAMPAIGN_IMAGE_SMALL_ICON_PUSH_KEY);
-        final Map<String, String> campaignAttributes = new HashMap<String, String>();
-
-        campaignAttributes.put(CAMPAIGN_ID_ATTRIBUTE_KEY, data.getString(CAMPAIGN_ID_PUSH_KEY));
-        campaignAttributes.put(CAMPAIGN_TREATMENT_ID_ATTRIBUTE_KEY, data.getString(CAMPAIGN_TREATMENT_ID_PUSH_KEY));
-        campaignAttributes.put(CAMPAIGN_ACTIVITY_ID_ATTRIBUTE_KEY, data.getString(CAMPAIGN_ACTIVITY_ID_PUSH_KEY));
-
-        this.pinpointContext.getAnalyticsClient().setCampaignAttributes(campaignAttributes);
-        log.info("Campaign Attributes are:" + campaignAttributes);
+        notificationChannelId = notificationDetails.getNotificationChannelId();
+        log.info("Event source Attributes are:" + eventSourceAttributes);
 
-        if (AWS_EVENT_TYPE_OPENED.equals(from)) {
-            return this.handleNotificationOpen(campaignAttributes, data);
+        //User clicked on the notification to open the app
+        if (eventSourceType.getEventTypeOpenend().equals(from)) {
+            return this.handleNotificationOpen(eventSourceAttributes, bundle);
         }
 
-        // Create the push event.
-        String eventType = null;
+        final AnalyticsEvent pushEvent;
         if (isAppInForeground) {
-            eventType = AWS_EVENT_TYPE_RECEIVED_FOREGROUND;
+            pushEvent = this.pinpointContext.getAnalyticsClient().createEvent(eventSourceType.getEventTypeReceivedForeground());
         } else {
-            eventType = AWS_EVENT_TYPE_RECEIVED_BACKGROUND;
+            pushEvent = this.pinpointContext.getAnalyticsClient().createEvent(eventSourceType.getEventTypeReceivedBackground());
         }
-        final AnalyticsEvent pushEvent = this.pinpointContext.getAnalyticsClient().createEvent(eventType);
-
-        // Add the campaign attributes.
-        addCampaignAttributesToEvent(pushEvent, campaignAttributes);
         pushEvent.addAttribute("isAppInForeground", Boolean.toString(isAppInForeground));
         try {
             // Ignore whether the app is in the foreground if the configuration indicates it should post
             // notifications in the foreground.
-            if (
-                !pinpointContext.getPinpointConfiguration().getShouldPostNotificationsInForeground() && isAppInForeground) {
+            if (!pinpointContext.getPinpointConfiguration().getShouldPostNotificationsInForeground() && isAppInForeground) {
                 // Notify the caller that the app was in the foreground.
-                return NotificationClient.CampaignPushResult.APP_IN_FOREGROUND;
+                return NotificationClient.PushResult.APP_IN_FOREGROUND;
             } else {
```

Comment:
If the push notification is received in the background, we also want to add these attributes to AnalyticsClient's global attributes. Should we call `addGlobalEventSourceAttributes` here?

Reply:
I thought we didn't want to count that since the app was already in the foreground and `ShouldPostNotificationsInForeground` is set to `false`. If the `ShouldPostNotificationsInForeground` is set to `true` if would take the else path of this if statement. 

Let me know if this makes sense or if I misunderstood you.

Reply:
Sorry, I meant the else case, which is when app is in the background.

Reply:
With regards to the concurrency issue when clearing the global attributes, I don't believe that's an issue for us. The global attributes are stored in an in-memory `ConcurrentHashMap`. When events are recorded, we pull those attributes from that data structure and create the event payload.